src/Type/Php/ArraySumFunctionDynamicReturnTypeExtension.php

```diff
@@ -36,12 +36,20 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		if ($arrayType->isIterableAtLeastOnce()->yes()) {
-			return TypeCombinator::intersect($itemType, new UnionType([new IntegerType(), new FloatType()]));
+			if ($itemType instanceof IntegerType || $itemType instanceof FloatType) {
```

Comment:
Better comparison would be:

`(new UnionType([new IntegerType(), new FloatType()]))->isSuperTypeOf($itemType)->yes()`